### PR TITLE
Added installation instructions for using LLVM 6.0 plus other small changes to previous installation instructions

### PR DESCRIPTION
**Pull request by ccadar**

### CODE REVIEW

**Comment by ccadar** [comment]
@251 , @MartinNowack , @kren1 If any of you has time to take a look at the instructions for 6.0, that would be great. I'd like to merge them tomorrow.

**Comment by kren1** [comment]
I wouldn't be recommending llvm 6 to people at this point as the optnone makes it really annoying.

You need to be really careful when compiling stuff with higher versions of llvm, because you run into strange issues with fortified functions and optnone, so I think the user experience is poor.

**Comment by ccadar** [comment]
@kren1 thanks for the feedback, but I haven't encountered any of these issues.  At this point, given that we've been using LLVM 6.0 successfully in a few projects, I would rather document/improve these issues, rather than still recommending an ancient LLVM version.

**Comment by ccadar** [comment]
Can you provide some examples regarding the fortified functions?

**Comment by ccadar** [comment]
With optnone, I think it's just a matter of documenting this.  Is there any evidence this will change in future versions?  If not, there's nothing to wait for, really :)

**Comment by kren1** [comment]
https://github.com/kren1/klee-docker/blob/llvm-7/miscprogs7.Dockerfile

This is what I need to compile the programs I use for llvm7, it was a ruined afternoon for me figuring this out and I doubt many people will go through the effort.

Regarding what is there to wait, we can start stirpping optnone in KLEE or provide an `wllvm` wrapper that compiles with the right flags.

For the fortified functions we need to.model them in the library but @MartinNowack said that he did it and it didn't get into klee for some reason.

**Comment by kren1** [comment]
These issues are not present in 3.8 yet so no meed to stick to 3.4 because of it.

**Comment by ccadar** [comment]
Hi @kren1 , I think it's possible to document this; the Coreutils tutorial would be a great start.  If you had time to help with this before the release, that would be great.  LLVM 3.8 is becoming unmaintainable on modern distributions (see e.g., https://github.com/klee/klee/pull/894#issuecomment-466623353), and it also makes it difficult to interact with other LLVM-based tools (as you know well).  Maintaining so many old versions of LLVM in the KLEE codebase is also becoming problematic. So I think it's important to stick with the plan to move the upcoming release to LLVM 6.0.

**Comment by kren1** [comment]
I find llvm 3.8 to be the most quite good at compatibility with various other tools and on that topic I believe that having klee compatible with a lot.of llvm versions helps a lot (3.4-3.7 is probably unnecessary). Also llvm 3.8 is the llvm for Ubuntu 16.04 LTS so I think it will be around for a while. In anycase I'll stop arguing this.

Btw why llvm 6.0? That's 2 years old by now and llvm 7 is working great. As a fun fact llvm 6 and 7 are (mostly) bitcode compatible.

**Comment by ccadar** [comment]
We'll keep support for 3.8.  I agree it's a good lower bound for now.

I don't mind recommending 7.0 if we're confident it works as well as 6.0 with KLEE.  We've been successfully using 6.0 in a couple of projects in the group (the artifact you've recently worked on uses 6.0 too!), which is why I favour it.

**Comment by kren1** [comment]
I've been using llvm 7 for my two artefacts, also I compiled test comp with 7.

File: build-llvm60.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+   ```bash
+   $ curl -OL https://github.com/google/googletest/archive/release-1.7.0.zip
+   $ unzip release-1.7.0.zip
+   ```
+
+   This will create a directory called `googletest-release-1.7.0`.
+
+6. **(Optional) Install lit:**
+
+   For testing the `lit` tool is used. If you LLVM from a build tree you can
```
Comment (251) [inline build-llvm60.md]: verb missing

File: build-llvm60.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+   to the KLEE git repository you cloned in step 7.
+   
+   ```bash
+   $ cd build
+   $ cmake <CMAKE_OPTIONS> <KLEE_SRC_DIRECTORY>
+   ```
+
+   `<CMAKE_OPTIONS>` are the configuration options. These are documented in [README-CMake.md](https://github.com/klee/klee/blob/master/README-CMake.md).
+
+   For example, if you want to built KLEE with STP, the POSIX runtime, klee-uclibc and unit testing then the
```
Comment (251) [inline build-llvm60.md]: build

File: build-llvm60.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+     -DKLEE_UCLIBC_PATH=<KLEE_UCLIBC_SOURCE_DIR> \
+     -DENABLE_UNIT_TESTS=ON \
+     -DGTEST_SRC_DIR=<GTEST_SOURCE_DIR> \
+     -DLLVM_CONFIG_BINARY=<PATH_TO_llvm-config-6.0> \
+     -DLLVMCC=<PATH_TO_clang-6.0> \
+     -DLLVMCXX=<PATH_TO_clang++-6.0>
+     <KLEE_SRC_DIRECTORY>
+   ```
+
+   Where `<KLEE_UCLIBC_SOURCE_DIR>` is the absolute path the klee-uclibc source tree,
```
Comment (251) [inline build-llvm60.md]: to the

File: build-llvm60.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+   `<GTEST_SOURCE_DIR>` is the absolute path to the Google Test source tree.
+
+   **NOTE 1:** You can simply type `cmake ..` to use the default options for KLEE (but note that these will not include support for uClibC and the POSIX runtime**
```
Comment (251) [inline build-llvm60.md]: runtime** - remove "**"

File: build-llvm60.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+   ```bash
+   $ make unittests
+   ```
+   
+   **NOTE:** You can run both the system and unit tests with `make check`
+
+9. **You're ready to go! Check the [Tutorials]({{site.baseurl}}/tutorials) page to try KLEE.**
+
+**NOTE:** For testing real applications (e.g. Coreutils), you may need to increase your system's open file limit (ulimit -n). Something between 10000 and 999999 should work. In most cases, the hard limit will have to be increased first, so it is best to directly edit the `/etc/security/limits.conf` file.<br/><br/>
```
Comment (251) [inline build-llvm60.md]: limits.conf is just for pam, maybe we should add `/etc/sysctl.conf `...

File: getting-started.md [hunk 6]
```diff
@@ -8,5 +8,6 @@ slug: getting-started
 There are three ways to get started with KLEE.
 
 * [Use our Docker images]({{site.baseurl}}/docker): this is the fastest way to get started.
-* [Build from source against LLVM 3.4]({{site.baseurl}}/build-llvm34): this is the current recommended version.
-* [Build from source against LLVM 3.8]({{site.baseurl}}/build-llvm38): this is still experimental
+* [Build from source against LLVM 6.0]({{site.baseurl}}/build-llvm60): this is the currently recommended version
+* [Build from source against LLVM 3.8]({{site.baseurl}}/build-llvm38): this version of LLVM is deprecated but still work with KLEE
```
[the remark this answers is not in the document]
Comment (ccadar) [reply getting-started.md]: OK, we'll do that and only deprecate <3.8 in the upcoming release.

**Comment by ccadar** [comment]
@251, thanks for the feedback, I think I addressed it all.  Ready to merge it now?

**Comment by kren1** [comment]
I kind of liked having build instructions one click away from the homepage. Could also be a neat trick to get people to use LLVM 6.0 since they wouldn't need to decide.

**Comment by 251** [comment]
@ccadar I'm fine with it.

**Comment by ccadar** [comment]
@kren1 We could have two links, one to the Docker instructions, and one to the recommended version.  I decided to go this way to maintain the links on a single page, but maybe it is a way to encourage people to use the recommended version.  @251 do you have a preference?

**Comment by 251** [comment]
- Use KLEE Docker image
- Installation instructions

**Comment by ccadar** [comment]
Thanks.  I went with @kren1 's suggestion and added a link to Docker and one to building from source with the recommended LLVM.   We could also add at the beginning of each installation instruction a link to the index of all installation pages (i.e. to Getting Started).

I'll merge it now.

**Comment by kren1** [comment]
@ccadar the link seems broken now

**Pull request merged**